docs/style-guidelines.md

```diff
@@ -1,20 +1,147 @@
+internal const float MIN_TIME = 0.1f;
+
+public static readonly int WAVE_ANIM_HASH = Animator.StringToHash("Wave");    // Static Read-only 
+private static readonly int DANCE_ANIM_HASH = Animator.StringToHash("Dance");
+
+public static bool IsOpen;                                                    // Static 
+internal static bool isAvtive;                                             
+
+public readonly List<int> CachedNumbers = new List<int>();                    // Read-only 
+protected readonly List<Section> sections = new List<Section>();
+
+public int PublicFieldA;                                                      // Public  
+[HideInInspector] public int SomeField02;
+[NonSerialized] public int SomeField03;
+
+[SerializedField] internal Animator animator;                                 // [SerializedField]'s 
+[SerializedField] private AnimationClip[] clips;  
+
+protected float cooldown;                                                     // internal-protected-private's
+private bool isVisible;                         
+```
+### Properties specifics
+  * static and readonly goes first (similar as for fields, see example above 👆)
+  * `get`-only and `set`-only properties goes last.
+  * Access modifiers for `set` considered to have higher priority than `get` (considered to be more exposed). 
+    * For example, property with `public set` and `private get` goes before `get`-only Property or Property with `private set` and `public get`
+    
+```csharp
+public bool Property1 { get; set; }                                        // get-set order
+public bool Property2 { private get; set; }
+public bool Property3 { get; private set; }
+public bool Property4 { get; }
+
+protected bool Property5 { get; set; }
+private bool Property6 { get; set; }
+```
+### **Methods specifics**
+  * Helper and supplementary methods which called from another method should be placed after method that calls it (in most cases it is `private` functions).
+```csharp
+// Note: indentaion for helper methods is used only for clarity. It is not a part of our formating style 
+public void Test1()                   // called by other class
+{
+    Test1Helper1();
+    Test1Helper2();
+}
+
+  private void Test1Helper1() => A();  // called by Test1() 
+    internal void A() { }              // called by Test1Helper1()
+  private void Test1Helper2() => B();  // called by Test1()
+    public void B() { }                // called by Test1Helper2()
+
+private void Awake() { }               // called by Unity
+```
+  * Not-helper methods should follow the order (where its helper methods follows previous rule and allowed to be placed in between of this order):
+    * constuctor
+    * destructor 
+    * `public`
+    * `public` with `[PublicApi]` and `[UsedImplicitly]` attributes
+    * Unity-callbacks 
+      * `Awake`, `Start`, `OnEnable`, `OnDisable`, `OnDestroy`,
+      * other callbacks (with respect to `Enter`-`Stay`-`Exit` order)
```

Comment:
wdyt about moving `Unity-callbacks` up to the same place as constructor/destructor?
1. because, similar to constructor/destructor, they are events related to the script lifecycle
2. it's easier to spot that we are looking at a `MonoBehaviour` script, since most of the time `: MonoBehaviour` can be easier to miss

Reply:
It was proposed to the team, but the majority chose `public` to go first :(